### PR TITLE
Rn entry 2021.12.0

### PR DESCRIPTION
**Pull request by susanharper-okta**
## Description:
- **What's changed?** Release Notes entry 2021.12.0
- **Is this PR related to a Monolith release?** Release Notes entry 2021.12.0

### Resolves:

* [OKTA-451228](https://oktainc.atlassian.net/browse/OKTA-451228)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md [hunk 1]
```diff
@@ -4,6 +4,61 @@ title: Okta Identity Engine API Products release notes 2021
...
 
+## December
+
+### Monthly release 2021.12.0
+
+| Change                                                                   | Expected in Preview Orgs |
+|--------------------------------------------------------------------------|--------------------------|
+| [Browser-like devices can now use the Device Authorization Grant flow](#browser-like-devices-can-now-use-the-device-authorization-grant-flow) | December 8, 2021 |
+| [Custom domains with Okta-managed certificates](#custom-domains-with-okta-managed-certificates) | December 8, 2021 |
+| [Device Authorization grant type is now GA in Production](#device-authorization-grant-type-is-now-ga-in-production) | November 3, 2021 |
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md:15]:
```suggestion
| [Device Authorization grant type is now GA in Production](#device-authorization-grant-type-is-now-ga-in-production) | December 8, 2021 |
```

File: packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md [hunk 2]
```diff
@@ -4,6 +4,61 @@ title: Okta Identity Engine API Products release notes 2021
...
+
+#### Custom domains with Okta-managed certificates
+
+When you customize an Okta URL domain, your Okta-hosted pages are branded with your own URL. [Okta-managed certificates](/docs/guides/custom-url-domain/main/#configure-a-custom-domain-through-okta-managed-certificates) automatically renew through a Let’s Encrypt integration, a free certificate authority. Okta-managed certificate renewals lower customer developer maintenance costs and reduce the high risk of a site outage when certificates expire. <!--OKTA-444104-->
+
+#### Device Authorization grant type is now GA in Production
+
+Advancements in internet technology have seen an explosion of smart devices and the Internet of Things. Consumers need to sign in to applications that run on these devices, but the devices either lack support for a web browser or have limited ability for input, such as smart TVs, car consoles, and thermostats. As a result, users resort to insecure authentication solutions that are error-prone and time-consuming.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md:32]: internet > Internet (uppercase?)
Accepted as lowercase or uppercase when used as an adjective. However, what is our Dev Docs convention? I looked for it in our style guide and couldn't find it.

File: packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md [hunk 3]
```diff
@@ -4,6 +4,61 @@ title: Okta Identity Engine API Products release notes 2021
...
+
+When there are applications that use Okta's subdomain and other applications that use the custom domain, the issuer validation breaks because the value is hard-coded to one domain or the other.
+
+With Dynamic Issuer Mode, the issuer value in minted tokens is dynamically updated based on the URL that is used to initiate the original authorize request. See [Client application settings](/docs/reference/api/apps/#settings-10). <!--OKTA-447358-->
+
+#### Error response updated for malicious IP address sign-in requests
+
+If you block suspicious traffic and [ThreatInsight](/docs/reference/api/threat-insight/) detects that the sign-in request comes from a malicious IP address, Okta automatically denies the user access to the organization. The user receives an error in response to the request. From the user’s perspective, the blocked request can’t be identified as the result of ThreatInsight having identified the IP address as malicious. <!--OKTA-434409-->
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md:46]:
```suggestion
If you block suspicious traffic and [ThreatInsight](/docs/reference/api/threat-insight/) detects that the sign-in request comes from a malicious IP address, Okta automatically denies the user access to the organization. The user receives an error in response to the request. From the user’s perspective, the blocked request can’t be identified due to ThreatInsight having identified the IP address as malicious. <!--OKTA-434409-->
```

File: packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md [hunk 4]
```diff
@@ -4,6 +4,61 @@ title: Okta Identity Engine API Products release notes 2021
...
+
+#### Upload Logo for Org deprecated
+
+The Upload Logo for Org endpoint (`api/v1/org/logo`) is deprecated. Use the [Upload Theme Logo](/docs/reference/api/brands/#upload-theme-logo) (`/api/v1/brands/${brandId}/themes/${themeId}/logo`) endpoint instead. <!--OKTA-432207-->
+
+#### Bugs fixed in 2021.12.0
+
+* When the [Device Authorization Grant flow](/docs/guides/device-authorization-grant/main/) was used, token inline hooks weren't called. (OKTA-445422)
+
+* When users signed in to Okta Identity Engine using the [Sign-In Widget](/code/javascript/okta_sign-in_widget/), the **Username** field wasn't set as a required field. (OKTA-391311)
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md:60]:
```suggestion
* When users signed in to the Okta Identity Engine by using the [Sign-In Widget](/code/javascript/okta_sign-in_widget/), the **Username** field wasn't set as a required field. (OKTA-391311)
```

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md [hunk 5]
```diff
@@ -1,6 +1,82 @@
...
+## December
+
+### Monthly release 2021.12.0
+
+| Change                                                                   | Expected in Preview Orgs |
+|--------------------------------------------------------------------------|--------------------------|
+| [Okta Sign-In Widget, version 5.14.0](#okta-sign-in-widget-version-5-14-0) | December 8, 2021 |
+| [Browser-like devices can now use the Device Authorization Grant flow](#browser-like-devices-can-now-use-the-device-authorization-grant-flow) | December 8, 2021 |
+| [Custom domains with Okta-managed certificates](#custom-domains-with-okta-managed-certificates) | December 8, 2021 |
+| [Device Authorization grant type is now GA in Production](#device-authorization-grant-type-is-now-ga-in-production) | November 3, 2021 |
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/release-notes/2021/index.md:13]:
```suggestion
| [Device Authorization grant type is now GA in Production](#device-authorization-grant-type-is-now-ga-in-production) | December 8, 2021 |
```

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md [hunk 6]
```diff
@@ -1,6 +1,82 @@
...
+| [IdP discovery supported for Device Authorization Grant flow](#idp-discovery-supported-for-device-authorization-grant-flow) | December 8, 2021 |
+| [List Schemas operation for the Log Streaming API now available](#list-schemas-operation-for-the-log-streaming-api-now-available) | December 8, 2021 |
+| [The word "source" is now allowed with custom application username formats](#the-word-source-is-now-allowed-with-custom-application-username-formats) | December 8, 2021 |
+| [Upload Logo for Org deprecated](#upload-logo-for-org-deprecated) | December 8, 2021          |
+| [User Types API and Mappings API support OAuth 2.0](#user-types-api-and-mappings-api-support-oauth-2-0) | December 8, 2021 |
+| [Bugs fixed in 2021.12.0](#bugs-fixed-in-2021-12-0)               | December 8, 2021          |
+
+#### Okta Sign-In Widget, version 5.14.0
+
+For details about this release, see the Okta [Sign-In Widget Release Notes](https://github.com/okta/okta-signin-widget/releases/tag/okta-signin-widget-5.14.0). For more information about the Widget, see the Okta [Sign-In Widget guide](/code/javascript/okta_sign-in_widget/). <!--OKTA-449551-->
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/release-notes/2021/index.md:25]:
```suggestion
See the Okta [Sign-In Widget Release Notes](https://github.com/okta/okta-signin-widget/releases/tag/okta-signin-widget-5.14.0) for more information about this release. See the Okta [Sign-In Widget guide](/code/javascript/okta_sign-in_widget/) for more information about the Widget. <!--OKTA-449551-->
```

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md [hunk 7]
```diff
@@ -1,6 +1,82 @@
...
+
+#### Custom domains with Okta-managed certificates
+
+When you customize an Okta URL domain, your Okta-hosted pages are branded with your own URL. [Okta-managed certificates](/docs/guides/custom-url-domain/main/#configure-a-custom-domain-through-okta-managed-certificates) automatically renew through a Let’s Encrypt integration, a free certificate authority. Okta-managed certificate renewals lower customer developer maintenance costs and reduce the high risk of a site outage when certificates expire. <!--OKTA-444104-->
+
+#### Device Authorization grant type is now GA in Production
+
+Advancements in internet technology have seen an explosion of smart devices and the Internet of Things. Consumers need to sign in to applications that run on these devices, but the devices either lack support for a web browser or have limited ability for input, such as smart TVs, car consoles, and thermostats. As a result, users resort to insecure authentication solutions that are error-prone and time-consuming.
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/release-notes/2021/index.md:37]: Same comment as above for "internet" in "internet technology".

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md [hunk 8]
```diff
@@ -1,6 +1,82 @@
...
+
+When there are applications that use Okta's subdomain and other applications that use the custom domain, the issuer validation breaks because the value is hard-coded to one domain or the other.
+
+With Dynamic Issuer Mode, the issuer value in minted tokens is dynamically updated based on the URL that is used to initiate the original authorize request. See [Client application settings](/docs/reference/api/apps/#settings-10). <!--OKTA-447358-->
+
+#### Error response updated for malicious IP address sign-in requests
+
+If you block suspicious traffic and [ThreatInsight](/docs/reference/api/threat-insight/) detects that the sign-in request comes from a malicious IP address, Okta automatically denies the user access to the organization. The user receives an error in response to the request. From the user’s perspective, the blocked request can’t be identified as the result of ThreatInsight having identified the IP address as malicious. <!--OKTA-434409-->
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/release-notes/2021/index.md:51]:
```suggestion
If you block suspicious traffic and [ThreatInsight](/docs/reference/api/threat-insight/) detects that the sign-in request comes from a malicious IP address, Okta automatically denies the user access to the organization. The user receives an error in response to the request. From the user’s perspective, the blocked request can’t be identified due to ThreatInsight having identified the IP address as malicious. <!--OKTA-434409-->
```

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md [hunk 9]
```diff
@@ -1,6 +1,82 @@
...
+
+The OAuth 2.0 [Device Authorization Grant flow](/docs/guides/device-authorization-grant/main/) now supports routing rules and IdP Discovery. <!--OKTA-425256-->
+
+#### List Schemas operation for the Log Streaming API now available
+
+API users can now [discover available log stream schemas](/docs/reference/api/schemas/#list-log-stream-schemas) for all log stream types by making a GET request to `/api/v1/meta/schemas/logStream`. The response is an array that contains the Log Stream Schema objects. <!--OKTA-439999-->
+
+#### The word "source" is now allowed with custom application username formats
+
+Custom application username formats set by the [Apps API](/docs/reference/api/apps/) can now include the word "source". <!--OKTA-443206-->
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/release-notes/2021/index.md:63]:
```suggestion
Custom application username formats that are set by the [Apps API](/docs/reference/api/apps/) can now include the word "source". <!--OKTA-443206-->
```

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md [hunk 10]
```diff
@@ -1,6 +1,82 @@
...
+
+#### Upload Logo for Org deprecated
+
+The Upload Logo for Org endpoint (`api/v1/org/logo`) is deprecated. Use the [Upload Theme Logo](/docs/reference/api/brands/#upload-theme-logo) (`/api/v1/brands/${brandId}/themes/${themeId}/logo`) endpoint instead. <!--OKTA-432207-->
+
+#### User Types API and Mappings API support OAuth 2.0
+
+The [User Types API](/docs/reference/api/user-types/) and [Mappings API](/docs/reference/api/mappings/) have been updated to support OAuth 2.0. You can grant access to the User Types API using the `okta.userTypes.manage` and `okta.userTypes.read` scopes. You can grant access to the Mappings API using the `okta.profileMappings.manage` and `okta.profileMappings.read scopes`. See [Scopes and supported endpoints](/docs/guides/implement-oauth-for-okta/main/#scopes-and-supported-endpoints). <!--OKTA-436385-->
```
Comment (christinechoy-okta) [inline packages/@okta/vuepress-site/docs/release-notes/2021/index.md:71]:
```suggestion
The [User Types API](/docs/reference/api/user-types/) and [Mappings API](/docs/reference/api/mappings/) have been updated to support OAuth 2.0. You can grant access to the User Types API by using the `okta.userTypes.manage` and `okta.userTypes.read` scopes. You can grant access to the Mappings API by using the `okta.profileMappings.manage` and `okta.profileMappings.read scopes`. See [Scopes and supported endpoints](/docs/guides/implement-oauth-for-okta/main/#scopes-and-supported-endpoints). <!--OKTA-436385-->
```

**Review by christinechoy-okta — APPROVED** [review]
LGTM!

File: packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md (hunk 1 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md:15]: > Hey Christine - we document when it goes GA to Preview - which was November 3. So, this is correct.
Screen Shot 2021-12-07 at 4 28 54 PM

File: packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md (hunk 2 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md:32]: @christinechoy-okta - if it's not in our style guides - then we follow Microsoft Manual of Style
Screen Shot 2021-12-07 at 4 28 54 PM

File: packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md (hunk 4 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md:60]: @christinechoy-okta - when first introducing Okta Identity Engine - you don't use articles. On subsequent uses, it's THE Identity Engine
Screen Shot 2021-12-07 at 4 32 39 PM
.

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md (hunk 5 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/release-notes/2021/index.md:13]: @christinechoy-okta - see my comment on the same blurb in the Classic RNs.

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md (hunk 6 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/release-notes/2021/index.md:25]: Are we changing the blurb that we reuse each time? This is what is in our wiki? I'm not opposed - but we need to update the [How to work on weekly and monthly API Release Notes](https://oktawiki.atlassian.net/wiki/spaces/DOC/pages/1707806413/How+to+work+on+weekly+and+monthly+API+Release+Notes) doc.
Screen Shot 2021-12-07 at 4 35 55 PM

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md (hunk 7 above)
Comment (susanharper-okta) [reply packages/@okta/vuepress-site/docs/release-notes/2021/index.md:37]: @christinechoy-okta same response as above

File: packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md (hunk 1 above)
Comment (christinechoy-okta) [reply packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md:15]: 👍

File: packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md (hunk 2 above)
Comment (christinechoy-okta) [reply packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md:32]: 👍

File: packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md (hunk 4 above)
Comment (christinechoy-okta) [reply packages/@okta/vuepress-site/docs/release-notes/2021-okta-identity-engine/index.md:60]: 👍

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md (hunk 5 above)
Comment (christinechoy-okta) [reply packages/@okta/vuepress-site/docs/release-notes/2021/index.md:13]: 👍

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md (hunk 6 above)
Comment (christinechoy-okta) [reply packages/@okta/vuepress-site/docs/release-notes/2021/index.md:25]: Oh right. Leave as is then.

File: packages/@okta/vuepress-site/docs/release-notes/2021/index.md (hunk 7 above)
Comment (christinechoy-okta) [reply packages/@okta/vuepress-site/docs/release-notes/2021/index.md:37]: 👍

**Review by christinechoy-okta — COMMENTED** [review]
I resolved your queries.

**Pull request merged**